### ISSUE TITLE
The new findpython seems to break resolving python version

### ISSUE THREAD

**Issue by j1g5awi** [pdm-project/pdm#2115]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

## Steps to reproduce

build my docker image: https://github.com/nonepkg/plugin-all4one/tree/master/docker

## Actual behavior

```
The exported requirements file is no longer cross-platform. Using it on other platforms may cause unexpected result.
STATUS: Resolving packages from lockfile...
python.use_venv is on, creating a virtualenv for this project...
Traceback (most recent call last):
  File "/usr/local/bin/pdm", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/pdm/core.py", line 290, in main
    return Core().main(args or sys.argv[1:])
  File "/usr/local/lib/python3.10/site-packages/pdm/core.py", line 208, in main
    raise cast(Exception, err).with_traceback(traceback) from None
  File "/usr/local/lib/python3.10/site-packages/pdm/core.py", line 203, in main
    self.handle(project, options)
  File "/usr/local/lib/python3.10/site-packages/pdm/core.py", line 157, in handle
    command.handle(project, options)
  File "/usr/local/lib/python3.10/site-packages/pdm/cli/commands/export.py", line 69, in handle
    candidates = resolve_candidates_from_lockfile(project, requirements.values())
  File "/usr/local/lib/python3.10/site-packages/pdm/cli/actions.py", line 138, in resolve_candidates_from_lockfile
    provider = project.get_provider(for_install=True)
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 426, in get_provider
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 426, in get_provider
    repository = self.get_repository(ignore_compatibility=ignore_compatibility)
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 394, in get_repository
    return cls(sources, self.environment, ignore_compatibility=ignore_compatibility)
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 280, in environment
    self._environment = self.get_environment()
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 259, in get_environment
    if self.config["python.use_venv"] and self.python.get_venv() is not None
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 149, in python
    self._python = self.resolve_interpreter()
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 232, in resolve_interpreter
    venv_path = self._create_virtualenv()
  File "/usr/local/lib/python3.10/site-packages/pdm/project/core.py", line 268, in _create_virtualenv
    path = venv_backend.create(
  File "/usr/local/lib/python3.10/site-packages/pdm/cli/commands/venv/backends.py", line 106, in create
    python_version=self.ident,
  File "/usr/local/lib/python3.10/site-packages/pdm/cli/commands/venv/backends.py", line 55, in ident
    return self._resolved_interpreter.identifier
  File "/usr/local/lib/python3.10/functools.py", line 981, in __get__
    val = self.func(instance)
  File "/usr/local/lib/python3.10/site-packages/pdm/cli/commands/venv/backends.py", line 40, in _resolved_interpreter
    if self.python or py_version.valid and self.project.python_requires.contains(py_version.version, True):
  File "/usr/local/lib/python3.10/site-packages/packaging/specifiers.py", line 905, in contains
    item = Version(item)
  File "/usr/local/lib/python3.10/site-packages/packaging/version.py", line 196, in __init__
    match = self._regex.search(version)
TypeError: expected string or bytes-like object
```

https://github.com/nonepkg/plugin-all4one/actions/runs/5595811305/jobs/10232083436?pr=27

## Expected behavior

no error

## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:
PDM version:
  2.8.0
Python Interpreter:
  /home/jigsaw/Documents/Program/nonepkg/nonebot-plugin-all4one/docker/.venv/bin/python (3.11)
Project Root:
  /home/jigsaw/Documents/Program/nonepkg/nonebot-plugin-all4one/docker
Local Packages:

{
  "implementation_name": "cpython",
  "implementation_version": "3.11.3",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "6.4.3-arch1-2",
  "platform_system": "Linux",
  "platform_version": "#1 SMP PREEMPT_DYNAMIC Sat, 15 Jul 2023 19:25:49 +0000",
  "python_full_version": "3.11.3",
  "platform_python_implementation": "CPython",
  "python_version": "3.11",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
Yeah, the prereleases don't work for current PDM, we should set the upper bound.

Now you can set `findpython<1` to fix it.

**Comment by abersheeran** [comment]
How to set `findpython<1` in setup-pdm action? [a2wsgi action](https://github.com/abersheeran/a2wsgi/actions/runs/5612517183/jobs/10270429331) crashed.

**Comment by frostming** [comment]
fixed on 2.8.1
